### PR TITLE
Improve performance for python addons using xbmcgui.ListItem

### PR DESCRIPTION
**Pull request by popcornmix**
## Description
Improve performance for python addons using xbmcgui.ListItem

## Motivation and Context
See: http://trac.kodi.tv/ticket/17304#comment:28

## How Has This Been Tested?
In last two Milhouse nightly builds.
Myself and trac ticket reported have measured significant performance increases

## Types of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by popcornmix** [comment]
Ping @FernetMenta 
https://github.com/xbmc/xbmc/pull/9748 introduced the performance regression.

I suspect a better solution to this issue will be found for Leia, but this is a low risk fix that could be backported to Krypton.

File: xbmc/Application.cpp [hunk 1]
```diff
@@ -296,6 +296,7 @@ CApplication::CApplication(void)
   , m_fallbackLanguageLoaded(false)
   , m_WaitingExternalCalls(0)
   , m_ProcessedExternalCalls(0)
+  , m_ProcessedExternalDecay(0)
```
Comment (FernetMenta) [inline xbmc/Application.cpp]: please use c++11 style for init of members

Comment (popcornmix) [inline xbmc/Application.cpp]: I've switched them all. Is that okay?

Comment (FernetMenta) [inline xbmc/Application.cpp]: I meant doing init by assignment in the header. that is much better than having to keep the right order in the constructor.

Comment (popcornmix) [inline xbmc/Application.cpp]: Ah like this?

Comment (FernetMenta) [inline xbmc/Application.cpp]: yes, thanks

**Comment by DaveTBlake** [comment]
Do you think the Python regression could also be the cause of music playback stuttering when returning back to the visualisation screen? User reports it clearly here http://forum.kodi.tv/showthread.php?tid=306568.  Happens with visualisation, but not genreal lib navigation.

I going to set  a build going for him test

**Comment by FernetMenta** [comment]
Completely different issue. Most likely related to the visualization. Note that what you call python regression would show effect in the opposite direction, means cure stuttering in a/v playback. It's the outside world like python and addons that disturbs the core.

**Comment by FernetMenta** [comment]
jenkins build and merge

**Comment by FernetMenta** [comment]
jenkins is out of order

**Comment by basrieter** [comment]
Will this also be merged into the Krypton branch and be in 17.1 ?

**Comment by popcornmix** [comment]
Hopefully. Keep an eye on https://github.com/xbmc/xbmc/pull/11660

**Comment by basrieter** [comment]
I guess I should wake up and should have looked better at the text 

> popcornmix referenced this pull request 6 hours ago
Open
Improve performance for python addons using xbmcgui.ListItem #11660 

Sorry for that.

**Comment by posti85** [comment]
We all hope is merged soon xD it's known our kodi addon are running very slow in our Raspberries. Thank popcornmix for the code and basrieter for the research!

**Comment by MartijnKaijser** [comment]
@popcornmix could you make a backport PR?

**Comment by rembound** [comment]
I have a twitch.tv plugin that displays the video stream in a windowed videowindow `xbmc.Player().play(streamurl, windowed=True)` and the chat in a list as an overlay. This list is continually updated with the latest chat messages `messagelist.addItem(xbmcgui.ListItem(line))`.

It always worked flawlessly on my RPI3, but since the Kodi 17 update, the video stutters when adding items to the list.

The following code is run in a thread and is a minimal case. Every second, a message is added to the list. Every few seconds, the video that is playing will stutter.
```
messagelist.addItem(xbmcgui.ListItem("the quick brown fox jumps over the lazy dog"))
listsize = messagelist.size()
if listsize > CHAT_MAXLINES:
    messagelist.removeItem(0)
messagelist.selectItem(messagelist.size()-1)

time.sleep(1)
```

Could this patch also be applied to fix performance issues during video playback?

**Comment by popcornmix** [comment]
As this has only just been merged then I assume your issue has nothing to do with this PR.
(this PR actually does nothing different when video is playing).

Open a [trac](http://trac.kodi.tv) ticket if you have found a regression on Krypton compared to Jarvis.

**Pull request merged**